drivers/video/vnc/vnc_fbdev.c

```diff
@@ -577,6 +577,64 @@ static inline int vnc_wait_start(int display)
   return OK;
 }
 
+/****************************************************************************
+ * Name: vnc_session_alloc
+ *
+ * Description:
+ *   Alloc vnc session and init essential member of it.
+ *
+ ****************************************************************************/
+
+static int vnc_session_alloc(void)
+{
+  FAR struct vnc_session_s *session;
+  int display;
+  int ret = OK;
+  FAR uint8_t *fb;
+
+  for (display = 0; display < RFB_MAX_DISPLAYS; display++)
```

Comment:
what's problem to allocate session in vnc_server?

Reply:
Touch/kbd inited before start vnc_server thread, thus the `g_vnc_sessions` is NULL before `vnc_start_server`.

Reply:
Ok, look like we should move register after vnc_wait_start?

Reply:
Yes, we should move touch/kbd driver after vnc_start_server or apply the second commit to init session in up_fbinitialize

Reply:
it's more natural to initialize touch/kbd after vnc_start_server.

Reply:
OK